trotters/trotters.py

```diff
@@ -0,0 +1,259 @@
+
+from redbot.core import checks, commands, Config, modlog
+from redbot.core.bot import Red
+from redbot.core.config import Group
+from redbot.core.commands import Context, Cog
+from redbot.core.utils.predicates import MessagePredicate
+
+try:
+    from redbot.core.commands import GuildContext
+except ImportError:
+    from redbot.core.commands import Context as GuildContext
+
+
+class Trotters(commands.Cog):
+    __author__ = "TheBluekr#2702"
+
+    def __init__(self, bot):
+        super().__init__()
+        self.bot = bot
+        self.logger = logging.getLogger("aurelia.cogs.trotters")
+        self.config = Config.get_conf(self, identifier=428038701, force_registration=True)
+
+        default_role = {"addable": False, "USERS": {}}
+
+        self.config.register_role(**default_role)
+
+    def cog_unload(self):
+        pass
+
+    async def initialize(self):
+        await self.register_casetypes()
+
+    @staticmethod
+    async def register_casetypes():
+        # register mod case
+        role_case = {
+            "name": "Role List Update",
+            "default_setting": True,
+            "image": "\N{PAGE FACING UP}",
+            "case_str": "roleupdate",
+        }
+        try:
+            await modlog.register_casetype(**role_case)
+        except RuntimeError:
+            pass
+
+    # Commands
+    @commands.group()
+    @commands.guild_only()
+    async def trotters(self, ctx: Context):
+        """Trotter's Pass settings"""
+        pass
+
+    @checks.admin_or_permissions(manage_roles=True)
+    @trotters.command(name="set")
+    async def set_role(self, ctx, role: discord.Role, enabled: bool):
+        """
+        Sets role to be addable/removable
+        """
+
+        if not role:
+            return await ctx.maybe_send_embed("No role supplied")
+
+        if not enabled:
+            if await self.config.role(role).USERS():
+                pred = MessagePredicate.yes_or_no(ctx)
+            await ctx.send(f"Found logs for role {role}, do you want to erase them?")
+            try:
+                await self.bot.wait_for("message", check=pred, timeout=30)
+            except asyncio.TimeoutError:
+                return await ctx.send("Timed out.")
+            if pred.result:
+                await self.config.role(role).USERS.set({})
+            await self.config.role(role).addable.set(False)
+            await ctx.tick()
+        else:
+            await self.config.role(role).addable.set(True)
+            await ctx.tick()
+
+    @checks.admin_or_permissions(manage_roles=True)
+    @trotters.command(name="roles")
+    async def view_roles(self, ctx):
+        """
+        Lists all roles which can be added/removed
+        """
+        enabled = list()
+        for role in ctx.guild.roles:
+            if await self.config.role(role).addable() and not role.is_default():
+                enabled.append(role.name)
+        await ctx.maybe_send_embed(f"List of addable roles: \n%s" % (", ".join(enabled)))
```

Comment:
You'll need to add `pagify` here in case the number of addable roles is greater than the character limit for the message.

`pages = pagify("\n".join(enabled))`
`await ctx.send("List of addable roles:")`
`for page in pages:
    await ctx.maybe_send_embed(page)`